### PR TITLE
Send Market listings as a paged embed

### PR DESCRIPTION
**Pull request by tanndlin**

### CODE REVIEW

**Comment by tanndlin** [comment]
image

File: src/services/message.service.ts [hunk 1]
```diff
@@ -39,6 +39,34 @@ export class MessageService {
...
+      (reaction: MessageReaction, user: User) =>
+        ['⬅️', '➡️'].includes(reaction.emoji.name) && user.id !== msg.author.id, //Only run if its not the bot putting reacts
+      { time: 1000 * 60 * 10 } //Listen for 10 Minutes
+    );
+
+    let pageIndex = 0;
+    collecter.on('collect', (reaction: MessageReaction) => {
+      msg.reactions.removeAll().then(async () => {
+        reaction.emoji.name === '➡️' ? pageIndex++ : pageIndex--;
+        msg.edit(pages[pageIndex]);
```
Comment (c650) [inline src/services/message.service.ts:56]: can be out of bounds

Comment (tanndlin) [reply src/services/message.service.ts:56]: It only sends the arrows if its [available.](https://github.com/joey-colon/lion/blob/65a5836da2702a98008c3c1c051ba068bbd1dd82/src/services/message.service.ts#L58-L63)
So the user cannot go past the bounds

Comment (c650) [reply src/services/message.service.ts:56]: nice sneaky invariant. what happens if two people react at the same time though?

Comment (c650) [reply src/services/message.service.ts:56]: in a way such that you could increment past the pages' bounds

**Review by c650 — APPROVED** [review]

**Pull request merged**